cmd/submit.go

```diff
@@ -36,23 +36,45 @@ If called with the name of an exercise, it will work out which
 track it is on and submit it. The command will ask for help
 figuring things out if necessary.
 `,
+	Args: cobra.MaximumNArgs(1),
 	RunE: func(cmd *cobra.Command, args []string) error {
-		usrCfg, err := config.NewUserConfig()
+		// Validate input before doing any other work
+		exercise, err := cmd.Flags().GetString("exercise")
 		if err != nil {
 			return err
 		}
 
-		cliCfg, err := config.NewCLIConfig()
+		trackId, err := cmd.Flags().GetString("track")
 		if err != nil {
 			return err
 		}
 
-		if len(args) == 0 {
-			cwd, err := os.Getwd()
-			if err != nil {
-				return err
-			}
-			args = []string{cwd}
+		files, err := cmd.Flags().GetStringSlice("files")
+		if err != nil {
+			return err
+		}
+
+		// Verify correct usage of flags/args
+		if len(args) == 0 && len(files) == 0 && !(exercise != "" && trackId != "") {
+			return errors.New("must use --exercise and --track together")
+		}
+
+		if len(args) > 0 && (exercise != "" || trackId != "" || len(files) > 0) {
+			return errors.New("can't use flags and directory arguments together")
```

Comment:
the combination of files and directories is handled below, so we can simplify this one and remove the part where we're checking files here.

For the error message, maybe something like:

> You are submitting a directory. We will infer the track and exercise from that. Please re-run the command again without the flags.